### PR TITLE
Minor issue with attempt to Profile from UnityClassLib

### PR DESCRIPTION
**Pull request by van800**
Only show "Unable to automatically discover correct Unity Editor to debug" for debug. Avoid showing it for Profile.

### CODE REVIEW

File: rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/run/configurations/UnityAttachToEditorRunConfiguration.kt [hunk 1]
```diff
@@ -92,6 +93,26 @@ class UnityAttachToEditorRunConfiguration(project: Project, factory: Configurati
...
+
+            val isClassLibraryProject = project.isUnityClassLibraryProject()
+            if (!project.isUnityProjectFolder() && (isClassLibraryProject == null || isClassLibraryProject)) {
+                throw RuntimeConfigurationError("Unable to automatically discover correct Unity Editor to debug")
+            }
+        }
+        super.checkRunnerSettings(runner, runnerSettings, configurationPerRunnerSettings)
+    }
+
     override fun checkSettingsBeforeRun() {
```
Comment (citizenmatt) [inline rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/run/configurations/UnityAttachToEditorRunConfiguration.kt:116]: This method could be deleted now.

File: rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/run/configurations/UnityAttachToEditorRunConfiguration.kt [hunk 2]
```diff
@@ -92,6 +93,26 @@ class UnityAttachToEditorRunConfiguration(project: Project, factory: Configurati
...
+        configurationPerRunnerSettings: ConfigurationPerRunnerSettings?
+    ) {
+        if (runner is DotNetDebugRunner) {
+            // If we're a class library project that isn't in a Unity project folder, we can't guess at the correct project
+            // to attach to, so throw an error and show the dialog. This value will be null until the backend has finished
+            // loading. However, because we're a Unity run configuration, we can safely assume we're a Unity project, and if
+            // we're not inside a Unity project folder, then we can't automatically attach, so throw an error and show the
+            // dialog
+
+            val isClassLibraryProject = project.isUnityClassLibraryProject()
```
Comment (citizenmatt) [inline rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/run/configurations/UnityAttachToEditorRunConfiguration.kt:108]: This needs a check for `pid != null` first. If we've previously attached to an editor, we try to reconnect. I think this logic is still true for class libraries.

File: rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/run/configurations/UnityAttachToEditorRunConfiguration.kt [hunk 3]
```diff
@@ -92,6 +93,26 @@ class UnityAttachToEditorRunConfiguration(project: Project, factory: Configurati
         return null
     }
 
+    override fun checkRunnerSettings(
```
Comment (citizenmatt) [inline rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/run/configurations/UnityAttachToEditorRunConfiguration.kt:96]: Does this method have the same result as `checkSettingsBeforeRun`? I seem to remember that `checkSettingsBeforeRun` would show the editor form, so we can choose an editor to attach to, while `checkRunnerSettings` throws an exception and shows it as a balloon notification. Have I remembered that right?

**Review by citizenmatt — CHANGES_REQUESTED** [review]

File: rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/run/configurations/UnityAttachToEditorRunConfiguration.kt (hunk 3 above)
Comment (van800) [reply rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/run/configurations/UnityAttachToEditorRunConfiguration.kt:96]: they both show the editor form for me.

**Review by citizenmatt — APPROVED** [review]

**Pull request merged**